### PR TITLE
Fix wrong unittest, check whether loading collection will cause OOM

### PR DESCRIPTION
**Pull request by yah01**
Signed-off-by: yah01 <yang.cen@zilliz.com>
#16067

### CODE REVIEW

**Comment by yah01** [comment]
rerun ut

File: internal/querynode/task_test.go [hunk 1]
```diff
@@ -634,10 +634,16 @@ func TestTask_loadSegmentsTask(t *testing.T) {
...
+				SegmentSize:  totalRAM,
 			},
 		}
+		// Reach the segment size that would cause OOM
+		for node.loader.checkSegmentSize(defaultCollectionID, task.req.Infos, 1) == nil {
+			task.req.Infos[0].SegmentSize *= 2
+		}
 		err = task.Execute(ctx)
 		assert.Error(t, err)
+		assert.Contains(t, err.Error(), "OOM")
```
Comment (congqixia) [inline internal/querynode/task_test.go:646]: Nit: maybe we can change the error to a customized error type in the future.

Comment (yah01) [reply internal/querynode/task_test.go:646]: Agree! Error handling in Golang is too "free"

**Pull request merged**